apps/web/src/pages/studio-onboarding/preview.tsx

```diff
@@ -0,0 +1,189 @@
+  const { currentUser } = useAuth();
+  const [tab, setTab] = useState('Preview');
+  const [content, setContent] = useState('');
+  const [subject, setSubject] = useState('');
+
+  const navigate = useNavigate();
+  const { integrations = [] } = useActiveIntegrations();
+  const integration = useMemo(() => {
+    return integrations.find((item) => item.channel === 'email' && item.primary) || null;
+  }, [integrations]);
+  const { mutateAsync: triggerTestEvent, isLoading } = useMutation(testTrigger);
+  const { templates, loading } = useTemplates({ pageSize: 1, areSearchParamsEnabled: false });
+
+  const template = useMemo(() => {
+    if (!templates) {
+      return null;
+    }
+
+    return templates[0];
+  }, [templates]);
+
+  const identifier = useMemo(() => {
+    if (template === null) {
+      return null;
+    }
+
+    return template.triggers[0].identifier;
+  }, [template]);
+
+  const stepId = useMemo(() => {
+    if (template === null) {
+      return null;
+    }
+
+    return (template.steps[0] as IStepVariant).uuid;
+  }, [template]);
+
+  const { mutateAsync, isLoading: previewLoading } = useMutation(
+    (data) => api.post('/v1/echo/preview/' + identifier + '/' + stepId, data),
+    {
+      onSuccess(data: any) {
+        setContent(data.outputs.body);
+        setSubject(data.outputs.subject);
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (!identifier || !stepId) {
+      return;
+    }
+
+    mutateAsync();
+  }, [identifier, mutateAsync, stepId]);
+
+  const onTrigger = async () => {
+    const to = {
+      subscriberId: currentUser._id,
+      email: currentUser.email,
+    };
+
+    try {
+      await triggerTestEvent({
+        name: identifier,
+        to,
+        payload: {
+          __source: 'onboarding-test-workflow',
+        },
+      });
+
+      navigate(ROUTES.STUDIO_ONBOARDING_SUCCESS);
+    } catch (e: any) {
+      Sentry.captureException(e);
+    }
+  };
+
+  if (loading) {
+    return null;
+  }
+
+  return (
+    <div
+      className={css({
+        width: '100dvw',
+        height: '100dvh',
+      })}
+    >
+      <Header active={1} />
+      <Background />
+      <div
+        className={vstack({
+          alignContent: 'center',
+          height: '100%',
+        })}
+      >
+        <div
+          className={css({
+            width: '880px',
+            zIndex: 1,
+            paddingTop: '100',
+          })}
+        >
+          <Tabs
+            withIcon={true}
+            value={tab}
+            onTabChange={(value) => {
+              setTab(value as string);
+            }}
+            menuTabs={[
+              {
+                icon: <IconVisibility />,
+                value: 'Preview',
+                content: (
+                  <PreviewWeb
+                    integration={integration}
+                    content={content}
+                    subject={subject}
+                    onLocaleChange={() => {}}
+                    locales={[]}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Should these be populated?‏

Reply:
As we do not provide translations in this default workflow we do not need to populate them more than like this because they are required props.